lib/MusicBrainz/Server/Controller/Role/Load.pm

```diff
@@ -107,6 +107,9 @@ role
         }
 
         if ($entity) {
+          if ($entity->gid != $id) {
```

Comment:
Just a few changes here:

```suggestion
            if ($id_is_guid && $entity->gid ne $id) {
```

(1) You need to include `$id_is_guid` in the check since `$entity` won't always have a `gid`, and (2) use `ne` since `!=` is for numeric comparisons in Perl.

Indentation was also misaligned, it should be 4 spaces per level rather than 2 (same for below).